### PR TITLE
CHTNW Adapter: initial release

### PR DESCRIPTION
**Pull request by chtdsp**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- 

- [ ] [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?

- [ ] Other

## Description of change

Add ```chtnwBidAdapter```

- contact email of the adapter’s maintainer [chtdsp@cht.com.tw](mailto:chtdsp@cht.com.tw)
- test parameters for validating bids:
```
bids: [{
    bidder: 'chtnw',
    params: {
        placementId: '38EL412LO82XR9O6'
    }
}]
```

### CODE REVIEW

File: modules/chtnwBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          uuid: chtnwId,
+          device: device,
+          version: {
+            prebid: '$prebid.version$',
+            adapter: '1.0.0',
+          },
+          site: {
+            numIframes: bidderRequest.numIframes,
+            isAmp: bidderRequest.isAmp,
+            pageUrl: bidderRequest.page,
```
Comment (patmmccann) [inline modules/chtnwBidAdapter.js:65]: bidderRequest.refererInfo.page I think?

 bidderRequest.refererInfo.ref for the field below

File: modules/chtnwBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            referrer: window.location.referrer,
+          },
+        },
+        options: {
+          contentType: 'application/json'
```
Comment (patmmccann) [inline modules/chtnwBidAdapter.js:70]: this will cause a preflight cors request, not recommended

File: test/spec/modules/chtnwBidAdapter_spec.js [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    const request = spec.buildRequests(bidRequests);
+
+    it('Returns POST method', function () {
+      expect(request.method).to.equal('POST');
+    });
+
+    it('Returns general data valid', function () {
+      let data = request.data;
+      expect(data).to.be.an('object');
```
Comment (patmmccann) [inline test/spec/modules/chtnwBidAdapter_spec.js:50]: this is really weak testing

**Comment by chtdsp** [comment]
Hi @patmmccann, we have updated please review again.

**Comment by MartianTribe** [comment]
Docs PR: https://github.com/prebid/prebid.github.io/pull/4016

File: modules/chtnwBidAdapter.js (hunk 1 above)
Comment (chtdsp) [reply modules/chtnwBidAdapter.js]: Fix the typo

File: modules/chtnwBidAdapter.js (hunk 2 above)
Comment (chtdsp) [reply modules/chtnwBidAdapter.js]: Remove content-type header to avoid cors.

File: test/spec/modules/chtnwBidAdapter_spec.js (hunk 3 above)
Comment (chtdsp) [reply test/spec/modules/chtnwBidAdapter_spec.js:50]: Add more data type validation.

**Comment by ChrisHuie** [comment]
@chtdsp i don't see you support advertiserDomain. Can you please include handling for adomain in this new adapter?

**Comment by chtdsp** [comment]
> @chtdsp i don't see you support advertiserDomain. Can you please include handling for adomain in this new adapter? Also pretty please pull in recent commits. This pr is a bit behind master 🙏

Hi @ChrisHuie, I merged recent commits and update the advertiserDomain support. Thnaks.

**Review by ChrisHuie — APPROVED** [review]

**Pull request merged**